### PR TITLE
Fix serialization of primitives

### LINKED ISSUE #9033: Backend based primitives are not serializable via dill
### Environment

- **Qiskit Terra version**:  0.22
- **Python version**: 3.7
- **Operating system**: Any

### What is happening?

Backend based primitives can't be saved via dill. Reference primitives can be loaded/saved. `QuantumInstance` is also serializable. This issue comes from Qiskit Machine Learning where quantum models can be saved to a file, then loaded to continue training or for inference on a real hardware.

### How can we reproduce the issue?

Run a script

```python
import dill
from qiskit import Aer
from qiskit.primitives import BackendSampler

sampler = BackendSampler(Aer.get_backend("aer_simulator"))
with open("sampler.dill", "wb") as f:
    dill.dump(sampler, f)

with open("sampler.dill", "rb") as f:
    sampler = dill.load(f)
```

### What should happen?

An exception is raised:
```
python __dill_primitives.py

Traceback (most recent call last):
  File "__dill_primitives.py", line 10, in <module>
    sampler = dill.load(f)
  File ".../envs/dev-qml/lib/site-packages/dill/_dill.py", line 313, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File ".../envs/dev-qml/lib/site-packages/dill/_dill.py", line 525, in load
    obj = StockUnpickler.load(self)
TypeError: __new__() missing 1 required positional argument: 'backend'
```

### Any suggestions?

No exceptions must be raised.

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Fix #9033

### CODE REVIEW

**Review by t-imamichi — DISMISSED** [review]

File: test/python/primitives/test_backend_estimator.py [hunk 1]
```diff
@@ -322,6 +323,13 @@ def test_no_max_circuits(self):
             self.assertEqual(len(result.metadata), k)
             np.testing.assert_allclose(result.values, target.values, rtol=0.2, atol=0.2)
 
+    @combine(backend=BACKENDS)
+    def test_serialization(self, backend):
+        """Test of serialize and deserialize"""
+        estimator = BackendEstimator(backend=backend)
+        serialization = pickle.dumps(estimator)
+        _ = pickle.loads(serialization)
+
```
[the remark this answers is not in the document]
Comment (ikkoham) [reply test/python/primitives/test_backend_estimator.py:332]: Thanks. I got it. I will remove them.

**Review by t-imamichi — APPROVED** [review]

**Comment by t-imamichi** [comment]
@adekusar-drl Can you try this PR with your code?

File: releasenotes/notes/fix-serialization-primitives-c1e44a37cfe7a32a.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixed an issue with the primitive classes, :class:`~.Sampler` and :class:`~.Estimator`,
+    where instances were not able to be serialized with ``pickle``.
```
Comment (mtreinish) [inline releasenotes/notes/fix-serialization-primitives-c1e44a37cfe7a32a.yaml:5]:
```suggestion
    Fixed an issue with the primitive classes, :class:`~.BackendSampler` and :class:`~.BackendEstimator`,
    where instances were not able to be serialized with ``pickle``. In general these classes are not guaranteed
    to be serializable as :class:`~.BackendV2` and :class:`~.BackendV1` instances are not required to be
    serializable (and often are not), but the class definitions of :class:`~.BackendSampler` and
    :class:`~.BackendEstimator` no longer prevent the use of ``pickle``.
```

**Review by mtreinish — COMMENTED** [review]
LGTM, just a small suggestion for clarity in the release

**Comment by adekusar-drl** [comment]
@t-imamichi  Looks good, thanks!

File: releasenotes/notes/fix-serialization-primitives-c1e44a37cfe7a32a.yaml (hunk 2 above)
Comment (ikkoham) [reply releasenotes/notes/fix-serialization-primitives-c1e44a37cfe7a32a.yaml:5]: Thank you very much!

**Review by t-imamichi — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]

**Pull request merged**